src/pke/examples/ckks-noise-flooding.cpp

```diff
@@ -0,0 +1,199 @@
+  and the second computation with NATIVEINT = 128, or vice versa, if they wish.
+ */
+
+#include "openfhe.h"
+
+using namespace lbcrypto;
+
+// Demo function for NOISE_FLOODING_DECRYPT mode in CKKS
+void CKKSNoiseFloodingDemo();
+
+/**
+ * We recommend putting part of the CryptoContext inside a function because
+ * you must make sure all parameters are the same, except EXECUTION_MODE and NOISE_ESTIMATE.
+ *
+ * @param cryptoParams Crypto parameters that already have their execution mode set (and noise estimate, if in EXEC_EVALUATION mode).
+ * @return the cryptoContext.
+ */
+CryptoContext<DCRTPoly> GetCryptoContext(CCParams<CryptoContextCKKSRNS>& cryptoParams);
+
+/**
+ * We recommend putting the encrypted computation you wish to perform inside a function because
+ * you have to perform it twice. In this example, we perform two multiplications and an addition.
+ *
+ * @param cryptoContext Crypto context.
+ * @param publicKey Public key for encryption.
+ * @return the ciphertext result. The first iteration will return a ciphertext that contains a noise measurement.
+ * The second iteration will return the actual encrypted computation.
+ */
+Ciphertext<DCRTPoly> EncryptedComputation(CryptoContext<DCRTPoly>& cryptoContext, PublicKey<DCRTPoly> publicKey);
+
+int main(int argc, char* argv[]) {
+    CKKSNoiseFloodingDemo();
+    return 0;
+}
+
+void CKKSNoiseFloodingDemo() {
+    // ----------------------- Setup first CryptoContext -----------------------------
+    // Phase 1 will be for noise estimation.
+    // -------------------------------------------------------------------------------
+    std::cout << "---------------------------------- PHASE 1: NOISE ESTIMATION ----------------------------------"
+              << std::endl;
+    CCParams<CryptoContextCKKSRNS> parametersNoiseEstimation;
+    // EXEC_NOISE_ESTIMATION indicates that the resulting plaintext will estimate the amount of noise in the computation.
+    parametersNoiseEstimation.SetExecutionMode(EXEC_NOISE_ESTIMATION);
+
+    auto cryptoContextNoiseEstimation = GetCryptoContext(parametersNoiseEstimation);
+
+    usint ringDim = cryptoContextNoiseEstimation->GetRingDimension();
+    std::cout << "CKKS scheme is using ring dimension " << ringDim << std::endl << std::endl;
+
+    // Key Generation
+    auto keyPairNoiseEstimation = cryptoContextNoiseEstimation->KeyGen();
+    cryptoContextNoiseEstimation->EvalMultKeyGen(keyPairNoiseEstimation.secretKey);
+
+    // We run the encrypted computation the first time.
+    auto noiseCiphertext = EncryptedComputation(cryptoContextNoiseEstimation, keyPairNoiseEstimation.publicKey);
+
+    // Decrypt  noise
+    Plaintext noisePlaintext;
+    cryptoContextNoiseEstimation->Decrypt(keyPairNoiseEstimation.secretKey, noiseCiphertext, &noisePlaintext);
+    noisePlaintext->SetLength(1);
+    double noise = noisePlaintext->GetCKKSPackedValue()[0].real();
```

Comment:
Could we use the precision data member instead? Using the first slot looks a little bit tricky.

Reply:
Switched it to use GetLogError() method.